offchain-modules/lib/src/dapp/indexer/ckb_indexer.rs

```diff
@@ -0,0 +1,214 @@
+use crate::dapp::indexer::db::{
+    create_ckb_to_eth_record, get_eth_to_ckb_record_by_outpoint, get_latest_ckb_to_eth_record,
+    is_ckb_to_eth_record_exist, update_eth_to_ckb_status, CkbToEthRecord,
+};
+use crate::transfer::to_eth::parse_ckb_proof;
+use crate::util::ckb_util::parse_cell;
+use crate::util::config::ForceConfig;
+use anyhow::{anyhow, Result};
+use ckb_jsonrpc_types::Uint128;
+use ckb_sdk::rpc::Transaction;
+use ckb_sdk::HttpRpcClient;
+use ckb_types::packed;
+use ckb_types::packed::{Byte32, OutPoint};
+use ckb_types::prelude::{Builder, Entity, Pack};
+use force_eth_types::eth_recipient_cell::ETHRecipientDataView;
+use force_eth_types::generated::basic::ETHAddress;
+use force_sdk::indexer::IndexerRpcClient;
+use log::info;
+use shellexpand::tilde;
+use sqlx::MySqlPool;
+
+pub struct CkbIndexer {
+    pub force_config: ForceConfig,
+    pub rpc_client: HttpRpcClient,
+    pub indexer_client: IndexerRpcClient,
+    pub db: MySqlPool,
+}
+
+impl CkbIndexer {
+    pub async fn new(
+        config_path: String,
+        db_path: String,
+        rpc_url: String,
+        indexer_url: String,
+    ) -> Result<Self> {
+        let config_path = tilde(config_path.as_str()).into_owned();
+        let force_config = ForceConfig::new(config_path.as_str())?;
+        let rpc_client = HttpRpcClient::new(rpc_url);
+        let indexer_client = IndexerRpcClient::new(indexer_url);
+        let db = MySqlPool::connect(&db_path).await?;
+        Ok(CkbIndexer {
+            force_config,
+            rpc_client,
+            indexer_client,
+            db,
+        })
+    }
+
+    pub async fn start(&mut self) -> Result<()> {
+        let record_option = get_latest_ckb_to_eth_record(&self.db).await?;
+        let mut ckb_height;
+        if record_option.is_some() {
+            ckb_height = record_option.unwrap().block_number;
+        } else {
+            ckb_height = self
+                .rpc_client
+                .get_tip_block_number()
+                .map_err(|e| anyhow!("failed to get ckb current height : {}", e))?;
+        }
```

Comment:
try [`map_or_else`](https://doc.rust-lang.org/1.49.0/std/option/enum.Option.html#method.map_or_else)

Reply:
if record_option.is_some(),  record_option.unwrap() always success. it seems no problem.